src/lambda/wizards/samDeployWizard.ts

```diff
@@ -666,11 +666,31 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
      */
     private hasImages: boolean = false
 
-    public constructor(private readonly context: SamDeployWizardContext, private readonly regionNode?: any) {
+    /**
+     * Initial treenode passed as a command arg to the "Deploy" command
+     * (typically, when invoked from the File Explorer context-menu).
+     */
+    private regionNode: any | undefined
+    /**
+     * Initial template.yaml path passed as a command arg to the "Deploy"
+     * command (typically, when invoked from the File Explorer context-menu).
+     */
+    private samTemplateFile: vscode.Uri | undefined
+
+    /**
+     *
+     * @param context
+     * @param commandArg Argument given by VSCode when the "Deploy" command was invoked from a context-menu.
+     */
+    public constructor(private readonly context: SamDeployWizardContext, commandArg?: any) {
         super()
-        // All nodes in the explorer should have a regionCode property, but let's make sure.
-        if (regionNode && Object.prototype.hasOwnProperty.call(regionNode, 'regionCode')) {
-            this.response.region = regionNode.regionCode
+        if (commandArg && commandArg.path) {
+            // "Deploy" command was invoked on a template.yaml file.
+            // The promptUserForSamTemplate() call will be skipped.
+            this.samTemplateFile = commandArg as vscode.Uri
+        } else if (commandArg && commandArg.regionCode) {
+            this.regionNode = commandArg
```

Comment:
`this.regionNode` doesn't appear to be used outside this scope. Is it necessary?

Reply:
it's used by existing code, e.g. `skipOrPromptRegion`

Reply:
Ah, I see my TS is already getting rusty. I didn't notice it was previously defined as a field in the constructor args.